Library/Homebrew/cli/args.rb

```diff
@@ -28,6 +28,14 @@ def options_only
             .map(&method(:to_cli_option))
             .select { |arg| arg.start_with?("-") }
       end
+
+      def flags_only
```

Comment:
Meant to comment on this before: can the `ARGV.flags_only ` method be removed entirely now? If so and if it has no usage in Homebrew/homebrew-core: please remove it. Thanks!

Reply:
It is being used in `build.rb` and `software_spec.rb` 
Do we leave `software_spec.rb` too towards the end ?

Reply:
Yeh, sounds good. Could you add a comment or similar (can be in another PR) in various places to make clear 1) that function shouldn't be used elsewhere and 2) just any brain dumping based on your current investigations?